shaders/src/shadowing.fs

```diff
@@ -167,16 +172,46 @@ float ShadowSample_VSM(const mediump sampler2DArray shadowMap,
  */
 
 // PCF sampling
-float shadow(const mediump sampler2DArrayShadow shadowMap,
-        const uint layer, const highp vec4 shadowPosition) {
+float shadow(const bool DIRECTIONAL,
+        const mediump sampler2DArrayShadow shadowMap,
+        const uint layer, const uint index, const uint cascade) {
+
+    highp vec4 shadowPosition;
+
+    // This conditional is resolved at compile time
+    if (DIRECTIONAL) {
+#if defined(HAS_DIRECTIONAL_LIGHTING)
+        shadowPosition = getCascadeLightSpacePosition(cascade);
+#endif
+    } else {
+#if defined(HAS_DYNAMIC_LIGHTING)
+        shadowPosition = getSpotLightSpacePosition(index);
+#endif
+    }
+
     highp vec3 position = shadowPosition.xyz * (1.0 / shadowPosition.w);
     // note: shadowPosition.z is in the [1, 0] range (reversed Z)
     return ShadowSample_PCF(shadowMap, layer, position);
 }
 
 // VSM sampling
-float shadow(const mediump sampler2DArray shadowMap,
-        const uint layer, const highp vec4 shadowPosition) {
+float shadow(const bool DIRECTIONAL,
+        const mediump sampler2DArray shadowMap,
+        const uint layer, const uint index, const uint cascade) {
```

Comment:
Can we make `index`/`cascade` into a single argument?

Reply:
Yes we could, but I thought it would be nice to keep them separate, in case we want to have several directional shadow in the future.